src/structures/Message.js

```diff
@@ -557,10 +561,18 @@ class Message extends Base {
    * @readonly
    */
   get deletable() {
-    return Boolean(
-      !this.deleted &&
-        (this.author.id === this.client.user.id ||
-          this.channel.permissionsFor?.(this.client.user)?.has(Permissions.FLAGS.MANAGE_MESSAGES)),
+    if (this.deleted) {
+      return false;
+    }
+    if (!this.guild) {
+      return this.author.id === this.client.user.id;
+    }
+    if (!this.channel.viewable) {
+      return false;
+    }
```

Comment:
```suggestion
    if (this.deleted || !this.channel?.viewable) {
      return false;
    }
    if (!this.guild) {
      return this.author.id === this.client.user.id;
    }
```
These can be combined, and channel is nullable

Reply:
DMChannel does not have a "viewable" property.
This is not combined.

Reply:
You'd have the same problem with the existing code then

Reply:
If "existing" is meaning b75073c, fixed at 0b6a55d.
If "existing" is meaning that code below, why?
https://github.com/discordjs/discord.js/blob/main/src/structures/Message.js#L560-L564

Reply:
Yeah that's fine. You should add some comments explaining why the ordering is such, its not intuitive that's whats happening, someone else may accidentally combine them in the future.

Reply:
Added. Thank you.